src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java

```diff
@@ -110,6 +114,14 @@ public boolean isApplicable(
                 @SuppressWarnings("rawtypes") Class<? extends AbstractProject> jobtype) {
             return true;
         }
+        @POST
+        public FormValidation doCheckMatlabCommand(@QueryParameter String matlabCommand) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);

```

Comment:
Is there any reason why we're checking this? I would expect that Jenkins should already handle the permissions of who can edit a pipeline without us needing to do that.

Reply:
Same as above the read permission is implied hence such static methods are susceptible for security issues.